libafl_bolts/src/llmp.rs

```diff
@@ -2266,38 +2273,31 @@ where
     where
         F: FnMut(ClientId, Tag, Flags, &[u8]) -> Result<LlmpMsgHookResult, Error>,
     {
-        #[cfg(feature = "std")]
-        let current_time = current_time();
         let mut new_messages = false;
         for i in 0..self.llmp_clients.len() {
             let client_id = self.llmp_clients[i].id;
             match unsafe { self.handle_new_msgs(client_id, on_new_msg) } {
                 Ok(has_messages) => {
-                    // See if we need to remove this client, in case no new messages got brokered, and it's not a listener
-                    #[cfg(feature = "std")]
-                    if !has_messages && !self.listeners.iter().any(|&x| x == client_id) {
-                        let last_msg_time = self.llmp_clients[i].last_msg_time;
-                        if last_msg_time < current_time
-                            && current_time - last_msg_time > self.client_timeout
-                        {
-                            self.clients_to_remove.push(i);
-                            #[cfg(feature = "llmp_debug")]
-                            log::info!("Client #{:#?} timed out. Removing.", client_id);
-                        }
-                    }
                     new_messages = has_messages;
                 }
-                Err(Error::ShuttingDown) => self.clients_to_remove.push(i),
+                Err(Error::ShuttingDown) => self.clients_to_remove.push(client_id.0),
                 Err(err) => return Err(err),
             }
         }
 
-        // After brokering, remove all clients we don't want to keep.
-        for i in self.clients_to_remove.iter().rev() {
-            let client_id = self.llmp_clients[*i].id;
-            log::info!("Client #{:#?} disconnected.", client_id);
-            self.llmp_clients.remove(*i);
+        let remove_cnt = self.clients_to_remove.len();
```

Comment:
Are we sure `clients_to_remove` is unique? Else the count will be off

Reply:
it is unique there's only one process that could ever send this "client exit" message and it only sends it once

Reply:
now it's hashset, and it's unique